crypto/x509/x_pubkey.c

```diff
@@ -263,9 +263,43 @@ int i2d_PUBKEY(const EVP_PKEY *a, unsigned char **pp)
     if (a->ameth != NULL && a->ameth->pub_encode != NULL
```

Comment:
maybe collapse this code into the new check below..

Reply:
The xpk should also be inside the a->ameth != NULL block

Reply:
Yeah, I just realised that